File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/inport/RequestTourBookingCommand.java
package com.dominikgaller.alpinebooking.booking.core.inport;

import java.time.LocalDate;

/**
 * Command carrying the input data for the RequestTourBooking use case.
 *
 * <p>Contains only primitive and standard-library types — no domain value objects.
 * Structural validation (non-blank, future date, count >= 1) is enforced by the
 * domain layer when the driver maps this command to domain types.
 *
 * <p>SDD: See {@code documentation/ports/request-tour-booking.inport.spec.md}, section 2.1.
 */
public record RequestTourBookingCommand(
        String tourId,
        LocalDate tourDate,
        int participantCount,
        String contactName,
        String contactEmail
) {}

File: src/main/kotlin/com/example/contractmanagement/booking/core/inport/command/RequestTourBookingCommand.kt
package com.example.contractmanagement.booking.core.inport.command

import java.time.LocalDate

/**
 * Input for the RequestTourBooking use case (UC01).
 *
 * Carries only standard-library types — no domain value objects. Structural validation
 * happens when the driver maps this command onto domain types, so the domain remains the
 * guard even if a future caller has no adapter in front of it
 * (`coding-style.definition.md` § 6.2).
 *
 * SDD: see `documentation/ports/request-tour-booking.inport.spec.md` § 2.1.
 */
data class RequestTourBookingCommand(
    val tourId: String,
    val tourDate: LocalDate,
    val participantCount: Int,
    val contactName: String,
    val contactEmail: String,
)
